CMSgov/onemac
# OY2-11159: Make user emails case insensitive [REQUIRES MIGRATION]

Story: https://qmacbis.atlassian.net/browse/OY2-11159
Endpoint: https://d2p8veht746yc8.cloudfront.net/

### Details

Allow users to sign in with whatever casing Okta recognizes, yet still be able to access OneMAC.

### Changes

- All user email addresses are now stored and queried in lower case

### Implementation Notes

- **This will require manual migration in ALL higher environments.** That means **develop**, **VAL**, and **PROD**, each immediately after deployment and before the service outage is lifted. All we will need to migrate are users' IDs in the user table to all lower case. In PROD that is 4 total records at the time of writing.
- Be warned that other than the _new_ login described in the Test Plan, all test logins are now all-lowercase (e.g. `stateadminactivemi`).

### Test Plan

1. Log in as `stateSUBMITTERactive` and verify that your view as an active State Submitter is unchanged.
2. View your user profile and verify that:
    - Your email is displayed as `statesubmitteractive@...`
    - It looks the same as before and all of your accesses are visible.
